lib/solarwinds_apm/support/otlp_endpoint.rb

```diff
@@ -0,0 +1,196 @@
+          if valid?(ENV['SW_APM_SERVICE_KEY'])
+            @token, @service_name = ENV['SW_APM_SERVICE_KEY'].to_s.split(':')
+          else
+            SolarWindsAPM.logger.warn { "SW_APM_SERVICE_KEY is invalid: #{mask_token(ENV['SW_APM_SERVICE_KEY'])}" }
+          end
+
+          ENV['OTEL_EXPORTER_OTLP_HEADERS'] = "authorization=Bearer #{@token}"
+        end
+
+        agent_enable = token_type != 'invalid'
+      end
+
+      @agent_enable = agent_enable
+      agent_enable
+    end
+
+    def valid?(_service_key)
+      # servicekey checker also works on service name, may need to remove that part
+      service_key_checker = SolarWindsAPM::ServiceKeyChecker.new('ssl', false)
+      service_key_name = service_key_checker.read_and_validate_service_key
+      service_key_name != ''
+    end
+
+    def determine_lambda_env
+      if ENV['LAMBDA_TASK_ROOT'].to_s.empty? && ENV['AWS_LAMBDA_FUNCTION_NAME'].to_s.empty?
+        false
+      else
+        SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] lambda environment - LAMBDA_TASK_ROOT: #{ENV.fetch('LAMBDA_TASK_ROOT', nil)}; AWS_LAMBDA_FUNCTION_NAME: #{ENV.fetch('AWS_LAMBDA_FUNCTION_NAME', nil)}" }
+        true
+      end
+    end
+
+    # token and endpoint also need to be considered for get settings
+    # endpoint is for get settings, populate to OTEL_EXPORTER_OTLP_METRICS_ENDPOINT at the end for otlp related exporter
+    # three sources: otlp env variable, sw env variable, sw config file
+
+    # sampler_config = {
+    #         collector: "https://#{ENV.fetch('SW_APM_COLLECTOR', 'apm.collector.na-01.cloud.solarwinds.com')}:443",
+    #         service: service_key_name[1],
+    #         headers: "Bearer #{service_key_name[0]}",
+    #         tracing_mode: SolarWindsAPM::Config[:tracing_mode],
+    #         trigger_trace_enabled: SolarWindsAPM::Config[:trigger_tracing_mode],
+    #         transaction_settings: SolarWindsAPM::Config[:transaction_settings]
+    #       }
+    def configure_otlp_endpoint(data_type)
+      # for staging, our purpose, just use OTEL_EXPORTER_OTLP_METRICS_ENDPOINT directly
+      # https://otel.collector.cloud.solarwinds.com:443/v1/traces
+      # SW_ENDPOINT_REGEX = /^apm\.collector(?:\.[a-z0-9-]+)?\.cloud\.solarwinds\.com$/
+
+      return unless %w[TRACES METRICS LOGS].include?(data_type)
+
+      data_type_upper = data_type.upcase
+      data_type = data_type.downcase
+      endpoint_type = if ENV["OTEL_EXPORTER_OTLP_#{data_type_upper}_ENDPOINT"]
+                        "#{data_type}_endpoint"
+                      elsif ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
+                        'general_endpoint'
+                      elsif ENV['SW_APM_COLLECTOR'].nil? && !@lambda_env
+                        'default_nil'
+                      elsif ENV['SW_APM_COLLECTOR'].to_s.match?(SW_ENDPOINT_REGEX)
+                        'apm_proto'
+                      else
+                        'invalid'
+                      end
```

Comment:
i find this naming of "endpoint_type" pretty confusing, makes it seem like we're trying to resolve an OTLP or apm-proto type, when what i _think_ this is doing is: resolve the OTLP endpoint based on the precedence of:

* OTEL config for exporter (general, or signal-specific) endpoint
* SW APM config for collector, derived as follows:
  * if apm.collector.*.*.solarwinds.com , then otel.collector.*.*.solarwinds.com 
* Fallback value is otel.collector.na-01.cloud.solarwinds.com 

How about renaming to something like "resolve_from" or "endpoint_source"

Reply:
Renamed